pandas/tests/generic/test_frame.py

```diff
@@ -287,3 +289,36 @@ def test_deepcopy_empty(self):
         empty_frame_copy = deepcopy(empty_frame)
 
         self._compare(empty_frame_copy, empty_frame)
+
+    def test_datetimeField_after_setitem(self):
+        # This test covers the unexpected behaviour of datetimeField when using
+        # setitem on another column as reported in issue #6942
+
+        start = pd.to_datetime("20140401")
+
+        df = pd.DataFrame(
+            index=pd.date_range(start, periods=1), columns=["timenow", "Live"]
+        )
+
+        df.at[start, "timenow"] = datetime.today()  # initial time.
+        time1 = df.at[start, "timenow"]
+
+        time.sleep(1)  # sleep time of 1 second in between assignments.
+
+        df.at[
+            start, "timenow"
+        ] = datetime.today()  # modified time before 'Live' column is set.
+        time2 = df.at[start, "timenow"]
+
+        df.Live = True  # setting the 'Live' column to True.
+
+        time.sleep(1)  # sleep time of 1 second in between assignments.
```

Comment:
These `sleep` calls are not needed.

Reply:
Some checks were failing with AssertionError, so I added those